docs/why-netdata/1s-granularity.md

```diff
@@ -0,0 +1,80 @@
+1. Time-series databases (prometheus, graphite, opentsdb, influxdb, etc) centralize all the metrics. At scale, these databases can easily become the bottleneck of the whole infrastructure.
+
+2. SaaS providers base their business models on centralizing all the metrics. On top of the time-series database bottleneck they also have increased bandwidth costs. So, massively supporting high resolution metrics, destroys their business model.
+
+The funny thing, is that since a couple of decades, the world has fixed this kind
+of scaling problems: instead of scaling up, scale out, horizontally. That is,
+instead of investing on bigger and bigger central components, decentralize the
+application so that it can scale by adding more smaller nodes to it.
+
+There have been many attempts to fix this problem for monitoring. But so far, all
+required centralization of metrics, which can only scale up. So, although the
+problem is somehow managed, it is still the key problem of all monitoring
+platforms and one of the key reasons for increased monitoring costs.
+
+Another important factor, is how resource efficient data collection can be when
+running per second. Most solutions fail to do it properly. The data collection agent is
+consuming significant system resources when running "per second", influencing the
+monitored systems and applications to a great degree.
+
+Last, per second data collection is a lot harder. In busy virtual environments,
+there is constantly a latency of about 100ms spread randomly to all data sources.
+If data collection is not implemented properly, this latency introduces a random
+error of +/- 10%, which is quite significant for a monitoring system.
+
+So, the monitoring industry fails to massively provide high resolution metrics, mainly for 3 reasons:
+
+1. Centralization of metrics, makes monitoring cost inefficient at that rate.
+2. Data collection needs optimization, otherwise it will significantly affect the monitored systems.
+3. Data collection is a lot harder, especially on busy virtual environments.
+
+## what netdata does?
+
+Netdata decentralizes monitoring completely. Each Netdata node is autonomous.
+It collects metrics locally, it stores them locally, it runs checks against them
+to trigger alarms locally, and provides an API for the dashboards to visualize them.
+This allows Netdata to scale to infinity.
+
+Of course, Netdata can centralize metrics when needed. For example, it is not
+practical to keep metrics locally on ephemeral nodes. For these cases, Netdata
+streams the metrics in real-time, from the ephemeral nodes to one or more
+non-ephemeral nodes nearby. This centralization is again distributed. On a large
+infrastructure there may be many centralization points.
+
+To eliminate the error introduced by data collection latencies on busy virtual
+environments, Netdata interpolates collected metrics. It does this using
+microsecond timings, per data source, offering measurements with an error rate
+of 0.00001%.
```

Comment:
Would be nice to have something backing up this claim.

Reply:
You mean about the 100ms latency or the netdata accuracy?

100ms latency, randomly spread over all proc data collection: https://docs.google.com/presentation/d/18C8bCTbtgKDWqPa57GXIjB2PbjjpjsUNkLtZEz6YK8s/edit#slide=id.g422e696d87_0_57

Netdata measures its accuracy, when in debug mode: https://github.com/netdata/netdata/blob/36199f449852f8077ea915a3a14a33fa2aff6d85/database/rrdset.c#L1070-L1099

So, it checks every point and also the total volume of collected metrics, and logs if accuracy goes above 0.0001% (3 zeros, not 4 - my mistake): https://github.com/netdata/netdata/blob/36199f449852f8077ea915a3a14a33fa2aff6d85/libnetdata/storage_number/storage_number.h#L85

I'll add more info to the document.